test/Microsoft.Azure.Devices.Edge.Test/Module.cs

```diff
@@ -12,20 +12,24 @@ namespace Microsoft.Azure.Devices.Edge.Test
 
     public class Module : ManualProvisioningFixture
     {
-        private const string DefaultSensorImage = "mcr.microsoft.com/azureiotedge-simulated-temperature-sensor:1.0";
+        private string GetTempSensorImage()
```

Comment:
I'd have this function return a `(string name, string image)` tuple so that tests can do:

```
var tempSensor = GetTempSensor();
// ...
    builder => { builder.AddModule(tempSensor.name, tempSensor.image); },
// ...
EdgeModule sensor = deployment.Modules[tempSensor.name];
```

Reply:
Side question: How do you know when to ` return by tuplex ` or using say ` GetTempSensor( out tempSensor ) ` ?

Reply:
I think you should generally prefer tuples. From the `out` [docs](https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/keywords/out-parameter-modifier): "Declaring a method with out arguments is a classic workaround to return multiple values. Beginning with C# 7.0, consider tuples for similar scenarios."

Reply:
I know you asked for a tuple but I'm giving a `tempSensor` class.

Reply:
Fact check: I'm giving you a tempSensor tuple as requested.